src/util/painterscope.h

```diff
@@ -0,0 +1,34 @@
+//
+// Created by Ferran Pujol Camins on 2019-06-22.
+//
+
+#pragma once
+
+#include <QPainter>
+
+// This class provides RAII style management of a QPainter properties.
+//
+// PainterScope will save the painter state on creation, and restore it
+// on destruction.
+class PainterScope final {
+  public:
+    PainterScope() = delete;
+    PainterScope(const PainterScope&) = delete;
+    PainterScope& operator=(const PainterScope&) = delete;
+
+    explicit PainterScope(QPainter* pPainter)
+            : m_pPainter(pPainter) {
+        if (m_pPainter) {
```

Comment:
A debug assertion should be sufficient here. Passing a null pointer is unexpected and a programming error. For additional safety you can use VERIFY_OR_DEBUG_ASSERT with a return statement.